x-pack/legacy/plugins/lens/public/editor_frame_plugin/merge_tables.test.ts

```diff
@@ -69,4 +70,38 @@ describe('lens_merge_tables', () => {
       }
     `);
   });
+
+  it('should handle this week now/w', () => {
+    const { dateRange } = mergeTables.fn(
+      {
+        type: 'kibana_context',
+        timeRange: {
+          from: 'now/w',
+          to: 'now/w',
+        },
+      },
+      { layerIds: ['first', 'second'], tables: [] },
+      {}
+    );
+
+    expect(
+      moment
+        .duration(
+          moment()
+            .startOf('week')
+            .diff(dateRange!.fromDate)
+        )
+        .asDays()
+    ).toEqual(0);
+
+    expect(
+      moment
+        .duration(
+          moment()
+            .endOf('week')
+            .diff(dateRange!.toDate)
+        )
+        .asDays()
+    ).toEqual(0);
+  });
```

Comment:
So you're not mocking moment, but you are asserting that the datemath logic is doing the exact thing. I think this is a fine test.